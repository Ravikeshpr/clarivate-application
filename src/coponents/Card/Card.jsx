import Button from "../button/Button";
export default function Card({ item, handleAddAsFavourite = () => {}, type }) {
    console.log("ffffffffffffff", item);
    return (
        <div className="card">
            <img src={item?.url} alt={item?.title} className="card-img" />
            <div className="title">
                <span>{item?.id}:</span>
                <span>{` ${item?.title}`}</span>
            </div>
            <div>
                {type === "list" && (
                    <div>
                        {!item.isFavourite ? (
                            <Button
                                onClick={() => handleAddAsFavourite(item)}
                                label="Add to favourite"
                            ></Button>
                        ) : (
                            <div className="favourite">Marked Favourite</div>
                        )}
                    </div>
                )}
            </div>
        </div>
    );
}
