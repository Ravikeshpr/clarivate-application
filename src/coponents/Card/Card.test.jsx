import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "./Card";

describe("Card", () => {
    test("Card should render without props", () => {
        const componentTree = renderer.create(<Card />).toJSON();

        expect(componentTree).toMatchSnapshot();
    });
    test("renders correct buttons", () => {
        const props = {
            item: [
                { id: "1", title: "title1", url: "url1", isFavourite: false },
            ],
            type: "list",
        };
        render(<Card item={props.item[0]} type={props.type} />);
        const cardLabel = screen.getByText("Add to favourite");
        expect(cardLabel).toBeInTheDocument();
    });
});
