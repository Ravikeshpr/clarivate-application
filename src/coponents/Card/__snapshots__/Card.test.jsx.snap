// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card should render without props 1`] = `
<div
  className="card"
>
  <img
    className="card-img"
  />
  <div
    className="title"
  >
    <span>
      :
    </span>
    <span>
       undefined
    </span>
  </div>
  <div />
</div>
`;
